shifthappens/tasks/imagenet_m/__init__.py

```diff
@@ -0,0 +1,180 @@
+        dataset_folder = os.path.join(self.data_root, "imagenet-m")
+        print("Downloading to ", dataset_folder)
+        tv_utils.download_url(
+            HUMAN_ACCURACY_JSON,
+            dataset_folder,
+            "human_accuracy.json",
+            "81d98aa82e570a3b2db98af5ccf87145",
+        )
+        tv_utils.download_url(
+            CLASS_INFO_JSON,
+            dataset_folder,
+            "class_info.json",
+            "50608bba554770c263cc8e23a2903c1c",
+        )
+        tv_utils.download_url(
+            IMAGENET_VALIDATION_LABELS,
+            dataset_folder,
+            "val_labels.txt",
+            "eaa6788f6a51fca5fb184f235d0c4e62",
+        )
+        if LOAD_GREEDYSOUPS_LOGITS:
+            tv_utils.download_url(
+                GREEDYSOUPS_JSON,
+                dataset_folder,
+                "greedysoups.npz",
+                "736b127e29091f01737c3e26a212ee0d",
+            )
+
+        human_accuracy = json.load(
+            open(os.path.join(dataset_folder, "human_accuracy.json"))
+        )
+
+        imagenet_class_info = json.load(
+            open(os.path.join(dataset_folder, "class_info.json"))
+        )
+
+        # Create mapping from wnid to cid.
+        self._wnid_to_cid = {}
+        for entry in imagenet_class_info:
+            self._wnid_to_cid[entry["wnid"]] = entry["cid"]
+
+        # Load validation targets.
+        self._labels_list = (
+            open(os.path.join(dataset_folder, "val_labels.txt"))
+            .read()
+            .strip()
+            .splitlines()
+        )
+
+        # This contains the list of the 68 files comprising the ImageNet-M task.
+        self._imagenet_m_2022_files = set(human_accuracy["imagenet_m"])
+
+        # Load mapping of filename to the set of correct multilabels.
+        self._imagenet_m_2022 = {}
+        for image_name, data in human_accuracy["initial_annots"].items():
+            if image_name not in self._imagenet_m_2022_files:
+                continue
+            correct_multi_labels = data.get("correct", [])
+            unclear_multi_labels = data.get("unclear", [])
+            correct_cids = [self._wnid_to_cid[x] for x in correct_multi_labels]
+            unclear_cids = [self._wnid_to_cid[x] for x in unclear_multi_labels]
+            entry = {
+                "correct_multi_labels": correct_cids,
+                "unclear_multi_labels": unclear_cids,
+            }
+            self._imagenet_m_2022[image_name] = entry
+
+        if LOAD_GREEDYSOUPS_LOGITS:
+            self._greedy_soups = load_npy_as_dict(
+                os.path.join(dataset_folder, "greedysoups.npz")
+            )
+        else:
+            # Create mapping from torch DataLoader order to filename
+            self._filename_to_pred_index = {}
+            iif = tv_datasets.ImageFolder(root=imagenet.ImageNetValidationData)
+            for i, (filename, _) in enumerate(iif.samples):
+                im_filename = filename.split("/")[-1]
```

Comment:
```suggestion
                im_filename = os.path.normpath(os.path.basename(filename))
```
I believe it is better to use `os.path.basename` instead of `split`, to address the backslash in the path.